powerstore/resource_volume_test.go

```diff
@@ -0,0 +1,852 @@
+					resource.TestCheckResourceAttr("powerstore_volume.volume_create_test", "size", "2.5"),
+					resource.TestCheckResourceAttr("powerstore_volume.volume_create_test", "capacity_unit", "GB")),
+			},
+			{
+				Config: VolumeParamsRename,
+				Check:  resource.ComposeTestCheckFunc(resource.TestCheckResourceAttr("powerstore_volume.volume_create_test", "name", "test_acc_cvol_updated")),
+			},
+		},
+	})
+}
+
+// Create Volume with size in int and capacity unit TB
+func TestAccVolume_CreateVolumeWithMBInInt(t *testing.T) {
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("Dont run with units tests because it will try to create the context")
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testProviderFactory,
+		Steps: []resource.TestStep{
+			{
+				Config: VolumeParamsWithMBInInt,
+				Check:  resource.ComposeTestCheckFunc(checkCreateVolume(t, testProvider, "test_acc_cvol_mb")),
+			},
+		},
+	})
+}
+
+// Create Volume with size in float and capacity unit TB
+func TestAccVolume_CreateVolumeWithTBInFloat(t *testing.T) {
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("Dont run with units tests because it will try to create the context")
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testProviderFactory,
+		Steps: []resource.TestStep{
+			{
+				Config: VolumeParamsWithTBInFloat,
+				Check:  resource.ComposeTestCheckFunc(checkCreateVolume(t, testProvider, "test_acc_cvol_tb_float")),
+			},
+		},
+	})
+}
+
+// Test to create volume with invalid high capacity unit, maximum volume size is 230 TB so PB is invalid
+func TestAccVolume_CreateVolumeWithPB(t *testing.T) {
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("Dont run with units tests because it will try to create the context")
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testProviderFactory,
+		Steps: []resource.TestStep{
+			{
+				Config:      VolumeParamsWithPB,
+				ExpectError: regexp.MustCompile("Invalid Attribute Value Match"),
+			},
+		},
+	})
+}
+
+// Test to create volume with invalid capacity unit, valid are MB, TB, GB
+func TestAccVolume_CreateVolumeWithInvalidCapUnit(t *testing.T) {
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("Dont run with units tests because it will try to create the context")
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testProviderFactory,
+		Steps: []resource.TestStep{
+			{
+				Config:      VolumeParamsWithInvalidCapUnit,
+				ExpectError: regexp.MustCompile("Invalid Attribute Value Match"),
+			},
+		},
+	})
+}
+
+// todo fix this test case, since it is not working with validation on capacity unit
```

Comment:
TODO committed is it fine?

Reply:
This is acceptance test and coverage is not impacted so can be taken up in next release.